_blog/2018/v1alpha3-routing.md

```diff
@@ -0,0 +1,347 @@
+  - labels:
+      version: v1
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: reviews-test-v2
+spec:
+  destination:
+    name: reviews
+  precedence: 2
+  match:
+    request:
+      headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+  route:
+  - labels:
+      version: v2
+```
+
+In `v1alph3`, we provide the same configuration in a single `VirtualService` resource:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews
+spec:
+  hosts:
+    - reviews
+  http:
+  - match:
+    - headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+    route:
+    - destination:
+        name: reviews
+        subset: v2
+  - route:
+    - destination:
+        name: reviews
+        subset: v1
+```
+
+As you can see, both of the rules for the `reviews` service are consolidated in one place, which at first may or may not
+seem preferable. However, if you look closer at this new model, you’ll see there are fundamental differences that make
+`v1alpha3` vastly more functional.
+
+First of all, notice that the destination service for the `VirtualService` is specified using a `hosts` field (repeated field,
+in fact) and is then again specified in a `destination` field of each of the route specifications. This is a very important
+difference from the previous model.
+
+A `VirtualService` describes the mapping between one or more user-addressable destinations to the actual destination workloads inside the mesh. In our example, they are the same, however, the user-addressed hosts can be any DNS
+names with optional wildcard prefix or CIDR prefix that will be used to address the service. This can be particularly
+useful in facilitating turning monoliths into a composite service built out of distinct microservices without requiring the
+consumers of the service to adapt to the transition.
+
+For example, the following rule allows users to address both the reviews and ratings services of the Bookinfo application
+as if they are parts of a bigger (virtual) service at `http://bookinfo.com/`:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: bookinfo
+spec:
+  hosts:
+    - bookinfo.com
+  http:
+  - match:
+    - uri:
+        prefix: /reviews
+    route:
+    - destination:
+        name: reviews
+  - match:
+    - uri:
+        prefix: /ratings
+    route:
+    - destination:
+        name: ratings
+  ...
+```
+
+The hosts of a `VirtualService` do not actually have to be part of the service registry, they are simply virtual
+destinations. This allows users to model traffic for virtual hosts that do not have routable entries inside the mesh.
+These hosts can be exposed outside the mesh by binding the `VirtualService` to a `Gateway` configuration for the same host
+(see [Gateway](#gateway), below).
+
+In addition to this fundamental restructuring, `VirtualService` includes several other important changes:
```

Comment:
I suggest moving this paragraph higher up to provide further context to the changes before showing configuration examples. Otherwise, the reader will be repeatedly perplexed while trying to read the configuration (e.g. "What is this `subset` thing? -> Cmd-F 'subset' enter-enter-enter...").  Overall, IMHO, the narrative in each section should be: 

- Describe problems with v1alpha1(2): now it is well described, redundant route rules and all
- Introduce new concepts for v1alpha3: that's where short descriptions of terms such as 'subset' should be mentioned
- Provide examples of the configuration changes (code excerpts with commentary)
- Close section by delving deeper into the reasons behind the change and expanding on its ontology